client.go

```diff
@@ -206,7 +206,12 @@ func (c *Client) Resume(state SMState) error {
 }
 
 func (c *Client) Disconnect() {
-	// TODO: Add a way to wait for stream close acknowledgement from the server for clean disconnect
+	// TODO : Wait for server response for clean disconnect
+	presence := stanza.NewPresence(stanza.Attrs{From: c.config.Jid})
+	presence.Type = stanza.PresenceTypeUnavailable
+	c.Send(presence)
+	c.SendRaw(stanza.StreamClose)
```

Comment:
This looks wrong: the transport already handles sending the closing stanza, since it is stream-specific. For websockets you must not send `</stream:stream>`.

Reply:
This is also missing a corner case there `c.transport` is nil: in that case you still try to send a stanza which is guaranteed to fail.

Reply:
Yes, this is not finished, I should not have merged it into master yet. I'm expecting to merge the full thing later today. 
Basically, I'm adding a way to ack the server response when the client calls "disconnect", so that we wait for the closing stream tag (if we disconnect before we get it, we lose miss some messages that were in transit for example), and for this I need to rework the parser.
I just wrote this to test with my chat example and commited it by mistake. Will fix.
Thanks !